habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -467,3 +467,48 @@ def step(self, ee_pos, **kwargs):
             self._sim.viz_ids["ee_target"] = self._sim.visualize_position(
                 global_pos, self._sim.viz_ids["ee_target"]
             )
+
+
+@registry.register_task_action
+class HumanoidJointAction(RobotAction):
+    def __init__(self, *args, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.num_joints = 19
+
+    def reset(self, *args, **kwargs):
+        super().reset()
+
+    @property
+    def action_space(self):
+        num_joints = self.num_joints
+        num_dim_transform = 16
+        # The action space is the number of joints plus 16 for a 4x4 transformtion matrix for the base
+        return spaces.Dict(
+            {
+                "human_joints_trans": spaces.Box(
+                    shape=(4 * (num_joints + num_dim_transform),),
+                    low=-1,
+                    high=1,
+                    dtype=np.float32,
+                )
+            }
+        )
+
+    def step(self, **kwargs):
+        new_pos_transform = kwargs["human_joints_trans"]
```

Comment:
Can you just change the signature of step to have human_joints_trans argument? This looks too hacky.

Reply:
This will also allow you to add a docstring explaining the all zeros trick used here.

Reply:
Addressed in latest commit